pkg/scheduler/framework/plugins/imagelocality/image_locality.go

```diff
@@ -63,7 +64,11 @@ func (pl *ImageLocality) Score(ctx context.Context, state *framework.CycleState,
 	}
 	totalNumNodes := len(nodeInfos)
 
-	score := calculatePriority(sumImageScores(nodeInfo, pod.Spec.Containers, totalNumNodes), len(pod.Spec.Containers))
+	scoredImages := sets.New[string]()
+	imageScores := sumImageScores(nodeInfo, pod.Spec.InitContainers, &scoredImages, totalNumNodes)
+	imageScores += sumImageScores(nodeInfo, pod.Spec.Containers, &scoredImages, totalNumNodes)
+
+	score := calculatePriority(imageScores, scoredImages.Len())
```

Comment:
We should we minus the number of containers with duplicated images here?

Reply:
I think it is needed, because InitContainer and Containers are started sequentially.

Reply:
This is still unclear to me, what I mean is we calculate the score as 
```
func calculatePriority(sumScores int64, numContainers int) int64 {
	maxThreshold := maxContainerThreshold * int64(numContainers)
	if sumScores < minThreshold {
		sumScores = minThreshold
	} else if sumScores > maxThreshold {
		sumScores = maxThreshold
	}

	return int64(framework.MaxNodeScore) * (sumScores - minThreshold) / (maxThreshold - minThreshold)
}
```

But the maxThreshold is agnostic in different nodes now for the `numContainers` passed-in diffs with each other. then the result may be unconvinced. Shouldn't we make it unchanged in calculating?